src/angular/projects/spark-core-angular/src/lib/components/sprk-link/sprk-link.component.ts

```diff
@@ -61,36 +61,63 @@ export class SparkLinkComponent implements OnInit {
   isExternal = false;
 
   ngOnInit() {
+    // Sets the default href if none provided
```

Comment:
Can we get a test where `href="someUndefinedProp"` to make sure angular doesn't cast `undefined` to "`undefined`", as that would break the parser here. It might not matter as `/undefined` is a valid url even it it breaks things. Probably not a deal breaker as TS is only at build time not runtime

Reply:
That's a good idea. Working on that now!

Reply:
It works as expected for when the value of undefined gets passed in and it allows 'undefined' to be passed in as the href value. Which, as you mentioned is what we want. So, I think we are good because we have to allow the route of /undefined. Added a test!